### PR TITLE
Issue1843 loan pattern

### PR DESCRIPTION
**Pull request by Shelly-Shih**
Loan Design Pattern

- Add the example of loan design pattern
- Issue: https://github.com/iluwatar/java-design-patterns/issues/1843

Pull request description

- Add a new loan design pattern

> For detailed contributing instructions see https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

### CODE REVIEW

**Comment by Shelly-Shih** [comment]
To maintainer, 
I just created a pull request. Please let me know what you think.
Many Thanks.

File: loan/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: Loan
+category: Behavioral
+language: en
+tag:
+- Financial Management
```
Comment (iluwatar) [inline loan/README.md:6]: Let's try to stick with the existing tags, if possible. See https://github.com/iluwatar/java-design-patterns/wiki/07.-Categories-and-Tags

File: loan/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+---
+
+## Intent
+Loan pattern is commonly used in financial system design. It helps ensure that loan objects are created and managed correctly, allowing for accurate calculation of interest, repayments, and other crucial financial metrics, without leading to data inconsistencies or inaccuracies.
+
```
Comment (iluwatar) [inline loan/README.md:11]: Please add "Explanation" section according to https://github.com/iluwatar/java-design-patterns/wiki/02.-Pattern-template

File: loan/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+Use the Loan pattern when:
+
+you are designing a system that handles various types of loans, such as personal loans, mortgages, auto loans, etc.
+there's a need to accurately track and manage loan lifecycle events like disbursement, repayments, and foreclosure.
+Do not use the Loan pattern when:
+
+the financial domain of your system is simple and doesn't need a dedicated structure for managing loans.
+the system has other primary focuses and loans are just a small aspect that can be managed without a dedicated pattern.
+
+## Real world examples
```
Comment (iluwatar) [inline loan/README.md:25]: Remove the empty heading

File: loan/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+language: en
+tag:
+- Financial Management
+---
+
+## Intent
+Loan pattern is commonly used in financial system design. It helps ensure that loan objects are created and managed correctly, allowing for accurate calculation of interest, repayments, and other crucial financial metrics, without leading to data inconsistencies or inaccuracies.
+
+## Class diagram
+![alt text](./etc/loan.png "Loan Pattern Diagram")
```
Comment (iluwatar) [inline loan/README.md:13]: Prefer to use urm-maven-plugin to generate the diagram

File: loan/src/main/java/com/iluwatar/loan/Main.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
+package com.iluwatar.loan;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+
```
Comment (iluwatar) [inline loan/src/main/java/com/iluwatar/loan/Main.java:6]: Here, above the main class (usually named App.java), please explain the pattern briefly. Describe how the code example implements the pattern.

File: loan/src/test/java/com/iluwatar/loan/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest {
```
Comment (iluwatar) [inline loan/src/test/java/com/iluwatar/loan/MainTest.java:17]: We usually have AppTest that runs the example and checks there are no errors. Then we have a set of unit tests.

File: loan/src/test/java/com/iluwatar/loan/MainTest.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+    private Path testFilePath;
+
+    @BeforeEach
+    public void setUp() {
+        String currentDirectory = System.getProperty("user.dir");
+        System.out.println("Current working directory: " + currentDirectory);
```
Comment (iluwatar) [inline loan/src/test/java/com/iluwatar/loan/MainTest.java:24]: Prefer to use a proper logger (from Lombok, for example)

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**